tests/migration_test.go

```diff
@@ -832,6 +832,49 @@ var _ = Describe("[Serial][rfe_id:393][crit:high][vendor:cnv-qe@redhat.com][leve
 				By("Deleting the VMI")
 				Expect(virtClient.VirtualMachineInstance(vmi.Namespace).Delete(vmi.Name, &metav1.DeleteOptions{})).To(Succeed())
 
+				By("Waiting for VMI to disappear")
+				tests.WaitForVirtualMachineToDisappearWithTimeout(vmi, 120)
+			})
+			It("should reject additional migrations on the same VMI if the first one is not finished", func() {
+				vmi := tests.NewRandomFedoraVMIWithGuestAgent()
+				vmi.Spec.Domain.Resources.Requests[k8sv1.ResourceMemory] = resource.MustParse(fedoraVMSize)
+
+				By("Starting the VirtualMachineInstance")
+				vmi = runVMIAndExpectLaunch(vmi, 240)
+
+				// Need to wait for cloud init to finish and start the agent inside the vmi.
+				tests.WaitAgentConnected(virtClient, vmi)
+
+				By("Checking that the VirtualMachineInstance console has expected output")
+				Expect(console.LoginToFedora(vmi)).To(Succeed())
+
+				By("Starting a first migration")
+				migration1 := tests.NewRandomMigration(vmi.Name, vmi.Namespace)
+				migration1, err = virtClient.VirtualMachineInstanceMigration(migration1.Namespace).Create(migration1)
+				Expect(err).To(BeNil())
+
+				// Successfully tested with 40, but requests start getting throttled above 10, which is better to avoid to prevent flakyness
+				By("Starting 10 more migrations expecting all to fail to create")
+				var wg sync.WaitGroup
+				for n := 0; n < 10; n++ {
+					wg.Add(1)
+					go func(n int) {
+						defer GinkgoRecover()
+						defer wg.Done()
+						migration := tests.NewRandomMigration(vmi.Name, vmi.Namespace)
+						_, err = virtClient.VirtualMachineInstanceMigration(migration.Namespace).Create(migration)
+						Expect(err).To(HaveOccurred(), fmt.Sprintf("Extra migration %d should have failed to create", n))
+						Expect(err.Error()).To(ContainSubstring(`admission webhook "migration-create-validator.kubevirt.io" denied the request: in-flight migration detected.`))
+					}(n)
```

Comment:
i can't say with certainty that this will always pass. For instance, if we're testing on a cluster where api requests are rate limited, then it's feasible that shotgunning 10 requests in parallel might add up to more total time than the migration, which would result in another migration getting through.

Here's a thought, the runnStressTest function could take a timeout, something like this. 
```
        runStressTest := func(vmi *v1.VirtualMachineInstance, vmsize string, stressTimeoutSeconds) {
                By("Run a stress test to dirty some pages and slow down the migration")
                stressCmd := fmt.Sprintf("stress --vm 1 --vm-bytes %sM --vm-keep --timeout %ds&\n", vmsize, stressTimeoutSeconds)
                Expect(console.SafeExpectBatch(vmi, []expect.Batcher{
                        &expect.BSnd{S: "\n"},
                        &expect.BExp{R: console.PromptExpression},
                        &expect.BSnd{S: stressCmd},
                        &expect.BExp{R: console.PromptExpression},
                }, 15)).To(Succeed(), "should run a stress test")

                // give stress tool some time to trash more memory pages before returning control to next steps
                if stressTimeoutSeconds < 15 { 
                        time.Sleep(time.Duration(stressTimeoutSeconds) * time.Second)
                } else {
                        time.Sleep(15 * time.Second)
                }
        }
```

Then this test could run the stress test with a short timeout to ensure that the fedora VM can't migration faster than the api server requests. 

thoughts?

Reply:
I meant to use that, but forgot, fixed now.
The upside of that omission is that we saw all the tests pass without the VMI being stressed, so flakyness is highly unlikely with stress added!